src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala

```diff
@@ -97,28 +108,85 @@ trait LocalFileSystemOperations {
+  private def stageWorkflowInputs(descriptor: WorkflowDescriptor): Try[HostInputs] = {
     val hostInputsPath = Paths.get(LocalBackend.hostExecutionPath(descriptor).toFile.getAbsolutePath, "workflow-inputs")
-    descriptor.actualInputs map {case(name, value) => name -> stageWdlValue(value, hostInputsPath)}
+    val attemptedStagedInputs = descriptor.actualInputs map { case(name, value) =>
+      val call = descriptor.namespace.resolve(name.split("\\.").dropRight(1).mkString(".")) match {
+        case Some(c:Call) => Some(c)
+        case _ => None
+      }
+      name -> stageWdlValue(call, value, hostInputsPath)
+    }
+    attemptedStagedInputs collect { case (k:String, f: Failure[_]) => k -> f } match {
+      case m: Map[FullyQualifiedName, Failure[_]] if m.nonEmpty =>
+        val errors = m.values map { _.exception.getMessage } mkString "\n"
+        Failure(new UnsupportedOperationException(s"Failures during localization:\n\n$errors"))
+      case _ =>
+        Success(attemptedStagedInputs map { case (k, v) => k -> v.get})
+    }
   }
 
-  private def stageInputArray(array: WdlArray, hostInputsPath: Path): WdlArray = array.map(stageWdlValue(_, hostInputsPath))
+  private def stageInputArray(call: Option[Call], array: WdlArray, hostInputsPath: Path): Try[WdlArray] = {
+    val attemptedStagedArray = array.value.map(stageWdlValue(call, _, hostInputsPath))
+    attemptedStagedArray collect { case f: Failure[_] => f } match {
+      case s:Seq[Failure[_]] if s.nonEmpty =>
+        val errors = s map { _.exception.getMessage } mkString "\n"
+        Failure(new UnsupportedOperationException(s"Failures during localization of array $array:\n\n$errors"))
+      case _ => Success(WdlArray(array.wdlType, attemptedStagedArray.map(_.get)))
+    }
+  }
 
-  private def stageWdlValue(value: WdlValue, hostInputsPath: Path): WdlValue = value match {
-    case w: WdlFile => stageWdlFile(w, hostInputsPath)
-    case a: WdlArray => stageInputArray(a, hostInputsPath)
-    case x => x
+  private def stageWdlValue(call: Option[Call], value: WdlValue, hostInputsPath: Path): Try[WdlValue] = value match {
+    case w: WdlFile => stageWdlFile(call, w, hostInputsPath)
+    case a: WdlArray => stageInputArray(call, a, hostInputsPath)
+    case x => Success(x)
   }
 
-  private def stageWdlFile(wdlFile: WdlFile, hostInputsPath: Path): WdlFile = {
+  private def stageWdlFile(call: Option[Call], wdlFile: WdlFile, hostInputsPath: Path): Try[WdlFile] = {
     val originalPath = Paths.get(wdlFile.value)
     val executionPath = hostInputsPath.resolve(originalPath.getFileName.toString)
+
+    def invalidLocalization(s: String)(call: Option[Call], originalPath: Path, executionPath: Path): Try[Unit] =
+      Failure(new UnsupportedOperationException(s"Unrecognized localization strategy: $s"))
+
+    val localizers = LocalizationStrategies map {
+      case "hard-link" => localizePathViaHardLink _
```

Comment:
I don't really totally disagree... Definitely I think at the very least it should `log.error` if it hits a borked string so the user isn't completely blind to the fact that they fat-fingered.  And I even agree that there are circumstances where an ADT representation might be nice... the reason why I disagree about adding it now is that the source code only contains the strings `hard-link` and `soft-link` once each so there's nothing to be gained right now.  Maybe if we start using these strings in other places we should at that time consider how this best be refactored?